packages/hierarchies-react/src/presentation-hierarchies-react/itwinui/TreeRenderer.tsx

```diff
@@ -45,16 +46,46 @@ export function TreeRenderer({ rootNodes, expandNode, localizedStrings, selectNo
     selectNodes: selectNodes ?? noopSelectNodes,
     selectionMode: selectionMode ?? "single",
   });
-
+  const parentRef = useRef<HTMLDivElement>(null);
   const flatNodes = useMemo(() => flattenNodes(rootNodes), [rootNodes]);
 
+  const virtualizer = useVirtualizer({
+    count: flatNodes.length,
+    getScrollElement: () => parentRef.current,
+    estimateSize: () => 28,
+    overscan: 5,
+  });
+
+  const items = virtualizer.getVirtualItems();
+
   return (
     <LocalizationContextProvider localizedStrings={localizedStrings}>
-      <Tree.Root style={{ height: "100%", width: "100%" }}>
-        {flatNodes.map((flatNode) => (
-          <TreeNodeRenderer {...treeProps} expandNode={expandNode} onNodeClick={onNodeClick} onNodeKeyDown={onNodeKeyDown} node={flatNode} key={flatNode.id} />
-        ))}
-      </Tree.Root>
+      <div style={{ height: "100%", width: "100%", overflowY: "auto" }} ref={parentRef}>
+        <Tree.Root style={{ height: virtualizer.getTotalSize(), width: "100%", position: "relative" }}>
```

Comment:
This needs `minHeight: "100%"` to make sure tree takes all space when there are less nodes than available space